### ISSUE TITLE
If I don't set the config at first, the config can not be changed after then

### ISSUE THREAD

**Issue by YangEfei** [glepnir/dashboard-nvim#263]
I think it maybe not a bug, however it will be confusing.
I installed dashboard by Packer:
```lua
use {
  'glepnir/dashboard-nvim',
  event = 'VimEnter',
  requires = {'nvim-tree/nvim-web-devicons'}
}
```
You can notice that I remove the part of `config`, because I have a separate profile to manage all the plugins.
Then I modify the separate profile as below:
```lua
local status, db = pcall(require, "dashboard")
if not status then
  vim.notify("没有找到 dashboard")
  return
end

db.setup({
  theme = 'hyper',
  config = {
    week_header = {
      enable = true,
    },
    shortcut = {
      { desc = ' Update', group = '@property', action = 'PackerSync', key = 'u' },
      {
        desc = ' Files',
        group = 'Label',
        action = 'Telescope find_files',
        key = 'f',
      },
      {
        desc = ' Projects',
        group = 'DiagnosticHint',
        action = 'Telescope projects',
        key = 'p',
      },
      {
        desc = ' Dotfiles',
        group = 'Number',
        action = 'Telescope dotfiles',
        key = 'd',
      },
    },
  },
})
```
It works well at the first time. Then after I execute the command `PackerSync`, it will update the `packer_cimpile.lua` since the config of it has `event = VimEnter` (I don't know if my judgment is accurate, I think it should be the reason for this configuration item).

And it will add some lines in `packer_cimpile.lua`:
```git
diff --git a/nvim/plugin/packer_compiled.lua b/nvim/plugin/packer_compiled.lua
index d321ff9..eda04c6 100644
--- a/nvim/plugin/packer_compiled.lua
+++ b/nvim/plugin/packer_compiled.lua
@@ -110,8 +110,10 @@ _G.packer_plugins = {
     url = "https://github.com/hrsh7th/cmp-vsnip"
   },
   ["dashboard-nvim"] = {
-    loaded = true,
-    path = "/home/yangyifei.yyf/.local/share/nvim/site/pack/packer/start/dashboard-nvim",
+    loaded = false,
+    needs_bufread = false,
+    only_cond = false,
+    path = "/home/yangyifei.yyf/.local/share/nvim/site/pack/packer/opt/dashboard-nvim",
     url = "https://github.com/glepnir/dashboard-nvim"
   },
   ["diffview.nvim"] = {
@@ -262,6 +264,13 @@ _G.packer_plugins = {
 }

 time([[Defining packer_plugins]], false)
+vim.cmd [[augroup packer_load_aucmds]]
+vim.cmd [[au!]]
+  -- Event lazy-loads
+time([[Defining lazy-load event autocommands]], true)
+vim.cmd [[au VimEnter * ++once lua require("packer.load")({'dashboard-nvim'}, { event = "VimEnter *" }, _G.packer_plugins)]]
+time([[Defining lazy-load event autocommands]], false)
+vim.cmd("augroup END")

 _G._packer.inside_compile = false
 if _G._packer.needs_bufread == true then
```

After that, when I try to update my own profile, it can not make any changes.
I tried different ways to update the config of dashboard, including add `config` as the README
```lua
use {
  'glepnir/dashboard-nvim',
  event = 'VimEnter',
  config = function()
    require('dashboard').setup {
      theme = 'hyper',
      config = {
        week_header = {
          enable = true,
        },
        shortcut = {
          { desc = ' Update', group = '@property', action = 'PackerSync', key = 'u' },
          {
            desc = ' Files',
            group = 'Label',
            action = 'Telescope find_files',
            key = 'f',
          },
          {
            desc = ' Projects',
            group = 'DiagnosticHint',
            action = 'Telescope projects',
            key = 'p',
          },
          {
            desc = ' Dotfiles',
            group = 'Number',
            action = 'Telescope dotfiles',
            key = 'd',
          },
        },
      },
    }
  end,
  requires = {'nvim-tree/nvim-web-devicons'}
}
```
However, it never worked properly.
And I tried to reinstall it, it can't work too.
At last, I removed the item `event = VimEnter` and reinstall again. The configure can work!
```lua
diff --git a/nvim/plugin/packer_compiled.lua b/nvim/plugin/packer_compiled.lua
index eda04c6..98c9922 100644
--- a/nvim/plugin/packer_compiled.lua
+++ b/nvim/plugin/packer_compiled.lua
@@ -110,6 +110,7 @@ _G.packer_plugins = {
     url = "https://github.com/hrsh7th/cmp-vsnip"
   },
   ["dashboard-nvim"] = {
+    config = { "\27LJ\2\n7\0\0\3\0\2\0\0046\0\0\0'\2\1\0B\0\2\1K\0\1\0\28plugin-config.dashboard\frequire\0" },
     loaded = false,
     needs_bufread = false,
     only_cond = false,
```
I found that it add the config item above into `packer_compiled.lua`, so I can update it randomly, no need to reinstall anymore.

In short, you cannot remove the `config` item during installation, otherwise it will make subsequent configuration updates impossible. I think it maybe a defects of Packer?  I can't be sure about this, maybe we can emphasize that we can't remove this item to avoid subsequent newbies stepping in the hole like I did.

**Comment by glepnir** [comment]
if you edit the `config` in `packer` you need recompile it to make it take effect. lazy.nvim  is easy than packer. you don't need compile every time.
